### DISCUSSION TITLE
How does one translate TARGETPLATFORM into aarch64?

### DISCUSSION THREAD

**Discussion by axblum** [docker/buildx#2001] [Q&A]
Given the following sample dockerfile, how does one translate the builtin ARG `TARGETARCH` into aarch64 as the artifacts are named using this as the architecture identifier? Ideally, I would be able to build the multiarch image using this command 
`docker build -f Dockerfile-t test --platform linux/arm64, linux/amd64`

```
FROM amazoncorretto:11
ARG TARGETARCH
ADD ${TARGETARCH} . 
```

**Comment by crazy-max** [comment]
You can use https://github.com/tonistiigi/xx for this and `xx-info march`:

```dockerfile
FROM amazoncorretto:11
COPY --from=tonistiigi/xx / /
ARG TARGETPLATFORM
RUN echo march=$(xx-info march)
```

But you will not be able to use this in `ADD` I think.

**Reply by axblum** [reply]
Do you think this would be worth putting in a feature request for? This would simplify a lot of dockerfiles.

**Reply by crazy-max** [reply]
We discuss about it here: https://github.com/moby/buildkit/issues/4168
